File: backend_php/signup.php
<?php
    require("./php_cors.php");
    require("./connection.php");

    $response = array();

    if($_SERVER["REQUEST_METHOD"]=="POST"){
        $firstname = $_POST['firstname'];
        $lastname = $_POST['lastname'];
        $email = $_POST['email'];
        $mobile = $_POST['mobile'];
        $password = $_POST['password'];

        $sql = "create table if not exists user(id int(10) auto_increment primary key,firstname varchar(50),lastname varchar(50),email varchar(50),mobile varchar(10),password varchar(50));";
        $conn->query($sql);

        $sql = "select * from user where email='$email';";
        $res = $conn->query($sql);

        if($res->num_rows>0){
            $response["ModalTitle"]="Email already registered...";
            $response["ModalBody"]="A User has already registered with the specified Email..Please try again with another Email...";
            http_response_code(200);
        }
        else{
            $sql = "insert into user values(NULL,'$firstname','$lastname','$email','$mobile','$password');";
            if($conn->query($sql)){
                $response["ModalTitle"]="Registration Successful...";
                $response["ModalBody"]="Registration has been completed successfully..Please Login...";
                http_response_code(201);
            }else{
                $response["ModalTitle"]="Registration Failed...";
                $response["ModalBody"]="Error occured during registration..Please try again later...";    
                http_response_code(200);
            }
        }

        echo json_encode($response);
    }

?>

File: src/router/routes/signup.go
package routes

import (
	"encoding/json"
	"fmt"
	"net/http"
	"router/sqldb"

	//package for mysql
	_ "github.com/go-sql-driver/mysql"
)

//SignupHandler POST API
func SignupHandler(res http.ResponseWriter,req *http.Request){

	SetHeaders(&res)
	response := make(map[string]string)

	if(sqldb.DB==nil){
		response["ModalTitle"]="Service Unavailable...";
		response["ModalBody"]="Signup Service is unavailable right now... Please try again later";
		res.WriteHeader(http.StatusServiceUnavailable)
		panic("503 Error")
	}

	defer func(){
		if r := recover(); r!=nil{
			json.NewEncoder(res).Encode(response);
		}
	}()

	req.ParseMultipartForm(0)

	firstname := req.Form.Get("firstname")
	lastname := req.Form.Get("lastname")
	email := req.Form.Get("email")
	mobile := req.Form.Get("mobile")
	password := req.Form.Get("password")

	stmt := "select * from sureify.user where email = '"+email+"';"
	rows,_ := sqldb.DB.Query(stmt)
	defer rows.Close()

	userExists := rows.Next()
	if(userExists){
		response["ModalTitle"] = "Email already registered..."
		response["ModalBody"] = "A User has already registered with the specified Email..Please try again with another Email..."
		res.WriteHeader(http.StatusForbidden)
		panic("403 Error")

	}else{
		stmt = "insert into sureify.user values(NULL,'"+firstname+"','"+lastname+"','"+email+"','"+mobile+"','"+password+"',NULL);";
		_, err := sqldb.DB.Query(stmt);
		if err!=nil{
			response["ModalTitle"]="Registration Failed..."
			response["ModalBody"]="Error occured during registration..Please try again later..."    
			res.WriteHeader(http.StatusForbidden)
			fmt.Println(err.Error())
			panic("403 Error")
		}else{
			response["ModalTitle"]="Registration Successful..."
			response["ModalBody"]="Registration has been completed successfully..Please Login..."                
			res.WriteHeader(http.StatusCreated)	
		}
	}
	json.NewEncoder(res).Encode(response);
}
